### PR TITLE
Add Android Auto Features

### PR DESCRIPTION
**Pull request by applehat**
**Preface: I am *not* very versed in Android Development, so if I did anything incredibly stupid please let me know.**

What this basically does is uses https://github.com/martoreto/aauto-sdk to Expands the Persistent Notification created by NSClient to display on the home screen of Android Auto.

du0nupexcai-ehy

### CODE REVIEW

**Comment by applehat** [comment]
I pushed an updating trying to fix that merge conflict, but again... sort of an Android Dev noob. /shrug

**Comment by MilosKozak** [comment]
Sorry I don't know how car apps exactly works but I don't understand why do you want all the AAPS code in the app just to display NS page?
Rendering of NS is a part of AAPS

**Comment by MilosKozak** [comment]
@AdrianLxM @jotomo

**Comment by applehat** [comment]
Initially I tried to mirror the main activity into the car activity, but I wasn't able to get that to work with my level of experience.

The main feature here is the persistent notifications being pushed to Android Auto so they are visible while driving (first photo).

Im not against ripping out the Nightscout display part - i intended it as a temporary measure for viewing NS while driving (if I needed a more detailed view then I could get from the notification alone).

Ideally it would just mirror the MainActivity but in Android Auto those displays are a different activity class so I'm not sure it's possible to reuse.

**Comment by applehat** [comment]
I'll take another look tomorrow (it's 1am now) and see if I can bring in the main status display to the car screen and get rid of the Nightscout WebView.

**Comment by applehat** [comment]
It doesn't appear that its "easy" at this point to pull in Phone-Screen fragments (such as the Overview plugin) into Car-Screen activities easily. For now I've just updated my branch to remove that integration all together (its not terribly helpful anyway).

Now all this PR does is push the Notification to the Car Screen as well.

**Comment by gregorybel** [comment]
results look really good, can't wait to use it. hope it Will be merged soon.

**Comment by applehat** [comment]
Are there any other concerns?

File: app/src/main/res/layout/activity_car_main.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (AdrianLxM) [inline app/src/main/res/layout/activity_car_main.xml]: @applehat what does this xml file do? It seems to be linked/used nowhere?

**Comment by applehat** [comment]
Yeah, accedentally left over from when I was trying to inplament a full
screen display. Not needed for just the notification display.

**Comment by applehat** [comment]
No idea. Haven't tested with any updates.

**Comment by hartlijn** [comment]
I have tested it @applehat on latest Android Auto and it works flawless.
This feature deserves to get included in the dev-branch, making it available for others as well who run a recent dev version.
@applehat ; could you please merge your feature into dev branch? Would be highly appreciated!

**Comment by MilosKozak** [comment]
can someone review this and bring up to date?

**Comment by gregorybel** [comment]
I can do if anybody want to do it

**Comment by gregorybel** [comment]
short update: i am working on this now but unfortunately the external sdk is no longer available so I need to find the official way doing this.

**Comment by gregorybel** [comment]
first success!

capture du 2018-11-10 23 27 15

still need to improved.

**Comment by MilosKozak** [comment]
is it working?
is this still worth to merge?

**Comment by hartlijn** [comment]
Yes the #1556 is

**Comment by Lispro2011** [comment]
Yes, it's working fine for me!

Am Di., 19. Feb. 2019, 19:10 hat Milos Kozak <notifications@github.com>
geschrieben:

**Pull request closed without merging**